src/lib.rs

```diff
@@ -37,53 +36,54 @@ impl Iceoryx2Transport {
     fn encode_uuri_segments(uuri: &UUri) -> Vec<String> {
         vec![
             uuri.authority_name.clone(),
-            Self::encode_hex_no_leading_zeros(uuri.uentity_type_id() as u32),
-            Self::encode_hex_no_leading_zeros(uuri.uentity_instance_id() as u32),
-            Self::encode_hex_no_leading_zeros(uuri.uentity_major_version() as u32),
-            Self::encode_hex_no_leading_zeros(uuri.resource_id() as u32),
+            Self::encode_hex(uuri.uentity_type_id() as u32),
+            Self::encode_hex(uuri.uentity_instance_id() as u32),
+            Self::encode_hex(uuri.uentity_major_version() as u32),
+            Self::encode_hex(uuri.resource_id() as u32),
         ]
     }
 
-    fn encode_hex_no_leading_zeros(value: u32) -> String {
+    fn encode_hex(value: u32) -> String {
         format!("{:X}", value)
     }
 
-    fn compute_service_name(message: &UMessage) -> Result<String, UStatus> {
+    fn compute_service_name(source: &UUri,sink: Option<&UUri>) -> Result<String, UStatus> {

```

Comment:
Have you run `cargo fmt`? I think it should catch things like this and other whitespace which looks to be a bit off.

```suggestion
    fn compute_service_name(source: &UUri, sink: Option<&UUri>) -> Result<String, UStatus> {

```

Reply:
this should be addressed now